extensions/ql-vscode/src/extension.ts

```diff
@@ -570,7 +570,17 @@ async function activateWithInstalledDistribution(
         token: CancellationToken,
         uri: Uri | undefined
       ) => {
-        const quickPickItems = dbm.databaseItems.map<DatabaseQuickPickItem>(dbItem => (
+        let filteredDBs = dbm.databaseItems;
+        // If possible, only show databases with the right language (otherwise show all databases).
+        const queryLanguage = await findLanguage(cliServer, uri);
+        if (queryLanguage) {
+          filteredDBs = dbm.databaseItems.filter(db => db.language === queryLanguage);
+          if (filteredDBs.length === 0) {
+            void helpers.showAndLogErrorMessage(`No databases found for language ${queryLanguage}. Please add a suitable database to your workspace.`);
+            return;
+          }
+        }
+        const quickPickItems = filteredDBs.map<DatabaseQuickPickItem>(dbItem => (
```

Comment:
Note that if the user has no databases and doesn't pick a language when prompted, we try to display this prompt with nothing in it. There should probably be an extra check to display an error in this case and return, similarly to what happens with the filtered list above.

Reply:
Woops, good point! Thanks 😄